iothub-manager/app/com/microsoft/azure/iotsolutions/iothubmanager/services/external/ConfigurationsHelper.java

```diff
@@ -0,0 +1,90 @@
+package com.microsoft.azure.iotsolutions.iothubmanager.services.external;
+
+import com.microsoft.azure.iotsolutions.iothubmanager.services.exceptions.InvalidInputException;
+import com.microsoft.azure.iotsolutions.iothubmanager.services.helpers.QueryConditionTranslator;
+import com.microsoft.azure.iotsolutions.iothubmanager.services.models.DeploymentServiceModel;
+import com.microsoft.azure.iotsolutions.iothubmanager.services.models.DeploymentType;
+import com.microsoft.azure.iotsolutions.iothubmanager.services.models.DeviceGroup;
+import com.microsoft.azure.sdk.iot.service.Configuration;
+import org.apache.commons.lang3.StringUtils;
+import play.libs.Json;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static play.libs.Json.fromJson;
+
+public class ConfigurationsHelper {
+
+    private static final String DEPLOYMENT_TYPE_LABEL = "Type";
+    private static final String DEPLOYMENT_NAME_LABEL = "Name";
+    private static final String DEPLOYMENT_GROUP_ID_LABEL = "DeviceGroupId";
+    private static final String DEPLOYMENT_GROUP_NAME_LABEL = "DeviceGroupName";
+    private static final String DEPLOYMENT_PACKAGE_NAME_LABEL = "PackageName";
+
+    public static final String RM_CREATED_LABEL = "RMDeployment";
+
+    public static Configuration toHubConfiguration(final DeploymentServiceModel deployment) throws InvalidInputException {
+        final String deploymentId = UUID.randomUUID().toString();
+        final Configuration configuration = new Configuration(deploymentId);
+
+        final String packageContent = deployment.getPackageContent();
+        final Configuration pkgConfiguration = fromJson(Json.parse(packageContent), Configuration.class);
+        configuration.setContent(pkgConfiguration.getContent());
+
+        final DeviceGroup deploymentGroup = deployment.getDeviceGroup();
+        final String dvcGroupQuery = deploymentGroup.getQuery();
+        final String query = QueryConditionTranslator.ToQueryString(dvcGroupQuery);
+        configuration.setTargetCondition(StringUtils.isNotBlank(query) ? query : "*");
+        configuration.setPriority(deployment.getPriority());
+        configuration.setEtag("");
+
+        if(configuration.getLabels() == null) {
+            configuration.setLabels(new HashMap<>());
+        }
+        final Map<String, String> labels = configuration.getLabels();
+
+        // Required labels
+        labels.put(DEPLOYMENT_TYPE_LABEL, deployment.getType().toString());
+        labels.put(DEPLOYMENT_NAME_LABEL, deployment.getName());
+        labels.put(DEPLOYMENT_GROUP_ID_LABEL, deploymentGroup.getId());
+        labels.put(RM_CREATED_LABEL, Boolean.TRUE.toString());
+
+        Map<String, String> systemMetrics = pkgConfiguration.getSystemMetrics().getQueries();
+        if (systemMetrics != null)
```

Comment:
should this if be here?

Reply:
Same as dotnet. Removed this as SystemMetrics cannot be set.